drm-ffi/src/mode.rs

```diff
@@ -453,18 +455,18 @@ pub fn set_plane(
     src_h: u32,
 ) -> Result<drm_mode_set_plane, Error> {
     let mut plane = drm_mode_set_plane {
-        plane_id: plane_id,
-        crtc_id: crtc_id,
-        fb_id: fb_id,
-        flags: flags,
-        crtc_x: crtc_x,
-        crtc_y: crtc_y,
-        crtc_w: crtc_w,
-        crtc_h: crtc_h,
-        src_x: src_x,
-        src_y: src_y,
-        src_w: src_w,
-        src_h: src_h,
+        plane_id,
+        crtc_id,
+        fb_id,
+        flags,
+        crtc_x,
+        crtc_y,
+        crtc_w,
+        crtc_h,
+        src_x,
+        src_y,
+        src_h,
+        src_w,
```

Comment:
Did we mess up the order here or did you change this by accident?

Reply:
This surprised me as well, but that's actually how it's written in the DRM struct. See https://elixir.bootlin.com/linux/latest/C/ident/drm_mode_set_plane:

```c
struct drm_mode_set_plane {
	__u32 plane_id;
	__u32 crtc_id;
	__u32 fb_id; /* fb object contains surface format type */
	__u32 flags; /* see above flags */

	/* Signed dest location allows it to be partially off screen */
	__s32 crtc_x;
	__s32 crtc_y;
	__u32 crtc_w;
	__u32 crtc_h;

	/* Source values are 16.16 fixed point */
	__u32 src_x;
	__u32 src_y;
	__u32 src_h;
	__u32 src_w;
};
```

`crtc_` is `w/h`, `src_` is `h/w` :shrug:. Not sure what's more readable, following the DRM struct or sensible order? It doesn't matter anyway because Rust requires the field name, but OTOH this very much looks like a C compound literal that _is_ based on order.